### PR TITLE
Rails 5.1 upgrade

### PR DESCRIPTION
**Pull request by Souravirus**
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [X] code is in uniquely-named feature branch and has no merge conflicts
* [X] PR is descriptively titled
* [ ] PR body includes `fixes #0000`-style reference to original issue #
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by Souravirus** [comment]
Hi @jywarren, I am currently  stuck on this error with OpenID:
`NoMethodError:         NoMethodError: undefined method 'length' for #<ActionController::Parameters:0x0000000ec966d0>
            app/controllers/openid_controller.rb:19:in 'index'
            test/integration/openid_test.rb:58:in `block in <class:LoginFlowTest>'`
I cannot trace out where is the location of the `length` method. It would be great if you help me out with this error. As I don't have much knowledge about OpenID.

**Comment by jywarren** [comment]
Hmm, the error leads to `decode_request`:

https://github.com/publiclab/plots2/blob/caf18c8360ad806d4a58c1aff7ef31f347014c74/app/controllers/openid_controller.rb#L19

Maybe a clue here: http://edgeapi.rubyonrails.org/classes/ActionController/Parameters.html

I might try to look for errors with `decode_request` on stack overflow, and esp. if things have changed in Rails 5 - i see here, for example: https://stackoverflow.com/questions/17240106/what-is-the-best-way-to-convert-all-controller-params-from-camelcase-to-snake-ca

and in general it looks like it's expecting an array or a string, but getting a Parameters object, which has no `length` property. If you are still stuck i can try digging deeper!

**Comment by Souravirus** [comment]
Thanks for the hint @jywarren, I am now trying to remove this error.

**Comment by Souravirus** [comment]
Open_id authentication is now working fine.

**Comment by jywarren** [comment]
awesome!

**Comment by jywarren** [comment]
Ooh wow! A bit easier than last one, huh? 😄

**Comment by Souravirus** [comment]
Yeah this was quite easy :smile:

**Comment by jywarren** [comment]
Shall we merge it then? :-)

**Comment by jywarren** [comment]
@publiclab/reviewers can someone else look through this? I'd like to merge and publish tomorrow but want to be sure. Thanks!

**Comment by jywarren** [comment]
Were you able to test it with the docs in https://github.com/publiclab/plots2/blob/master/doc/OPENID.md ?

**Comment by Souravirus** [comment]
Yeah we can merge it. Yeah I tested the open id authentication as is written in this link https://github.com/publiclab/plots2/blob/master/doc/OPENID.md and it's working fine. Thank you !!

**Comment by jywarren** [comment]
THANK YOU @Souravirus and @sagarpreet-chadha !!!

**Comment by jywarren** [comment]
This is now live!

File: app/controllers/openid_controller.rb [hunk 1]
```diff
@@ -15,20 +15,36 @@ class OpenidController < ApplicationController
...
+        'openid.sreg.nickname', 
+        'return_to', 'openid.claimed_id', 
+        'openid.identity', 'openid.mode', 
+        'openid.ns', 'openid.ns.sreg', 
+        'openid.realm', 'openid.return_to', 
+        'openid.sreg.required', 
+        'openid.trust_root', 
+        'openid.id_select', 
+        'openid.immediate', 
+        'openid.cancel_url').to_h
```
Comment (siaw23) [inline app/controllers/openid_controller.rb]: btw i think you should put this in a `before_action`, and make a method to return `permitted_params` but make it private. @Souravirus 

PS: sorry for late review. busy.

**Pull request merged**